_posts/2019-07-11-deploy-spring-boot-kubernetes.md

```diff
@@ -0,0 +1,135 @@
+
+## Where the magic happens: deploying your application on the Kubernetes cluster
+
+All Kubernetes configurations are written in YAML. This is because Kubernetes configuration files are meant to be easily readable by the human eye, and the Kubernetes team decided to use YAML instead of JSON.
+
+### Deployment
+
+Now that the Docker image is created, it can now be deployed on the Kubernetes cluster. 
+First a Deployment configuration file needs to be written, which will have the configuration on how the application should run.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myapp
+  labels:
+    app: myapp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: myapp
+  template:
+    metadata:
+      labels:
+        app: myapp
+    spec:
+      containers:
+        - name: myapp
+          image: repo/app:latest
+          ports:
+            - containerPort: 8080
+          env: 
+            - name: SPRING_PROFILES_ACTIVE
+              value: "test"
+            - name: DB_HOSTNAME
+              valueFrom:
+                secretKeyRef:
+                  name: myapp-secret
+                  key: database-host
+            - name: DB_USER
+              valueFrom:
+                secretKeyRef:
+                  name: myapp-secret
+                  key: database-user
+            - name: DB_PASS
+              valueFrom:
+                secretKeyRef:
+                  name: myapp-secret
+                  key: database-pass
+            - name: DB_NAME
+              valueFrom:
+                secretKeyRef:
+                  name: myapp-secret
+                  key: database-name
+```
+There's a lot going on here which will be explained step by step.
+The first 2 lines are telling which Kubernetes API version is being used and what kind of Kubernetes [object](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/){:target="_blank" rel="noopener noreferrer"} that is being applied. As we want to create a new Deployment, we use the Deployment object (easy, right?).
+Lines 3 to 7 are just basic metadata tags so the developer knows which application (s)he is working with. This does not affect the behavior of the application in any way.
+
+Lines 8 to 13 are specifying how the container should be made and which image it should run. This is the image that was created with the Dockerfile earlier in this post. After that, the port that the container should listen to is mentioned, which is 8080 in this case. The replica value specifies how many 'instances' (also called [Pods](https://kubernetes.io/docs/concepts/workloads/pods/pod/){:target="_blank" rel="noopener noreferrer"}) that should be running. If the application is expecting a lot of requests, it might be useful to declare a higher number of replicas instead of 1.
```

Comment:
```suggestion
Lines 8 to 13 are specifying how the container should be made and which image it has to run. This is the image that we created with the Dockerfile earlier in this post. After that, it describes the port that the container should listen to, which is 8080 in this case. The replica value specifies how many 'instances' (also called [Pods](https://kubernetes.io/docs/concepts/workloads/pods/pod/){:target="_blank" rel="noopener noreferrer"}) that should be running. If the application is expecting a lot of requests, it might be useful to declare a higher number of replicas instead of 1.
```